Cannot compile Darktable on Fedora 30 beta (GCC9 / OpenMP 4.0)

**Describe the bug**

Darktable from master, as well as the latest stable release (2.6.2).

**To Reproduce**

1. Use Fedora 30 with GCC9
2. Try to compile; get an OpenMP compile error
3. Attempt to compile again without OpenMP (`--disable-openmp`); get a compiler error

I've tried compiling on another machine I've also upgraded to the Fedora 30 beta, and also compiling 2.6.2 as well — and was met with the same results with those other attemps.

**Expected behavior**

Darktable compilation was working fine in Fedora 29. Only after upgrading to 30 (currently frozen in beta) did Darktable stop compiling properly.

**Platform:**

 - Darktable Versions: master branch & 2.6.2
 - OS: Fedora 30 beta
 - CPU: Intel Core i7-7500U
 - GPU: Intel HD Graphics 620 (rev 02)

**Additional context**

Fedora 30 switched to GCC9 & OpenMP 4.0 — and this changes things a bit:
https://gcc.gnu.org/gcc-9/porting_to.html

Rawspeed recently updated to support GCC9 / OMP4 (and compiles fine on F30):
https://github.com/darktable-org/rawspeed/commit/c82d0f30a8f8865592013f0156c5e745ae41c250

**Example of compile errors**

Standard recommend compile with `./build.sh --prefix /usr/local --build-type Release`:
```
[ 26%] Building C object src/CMakeFiles/lib_darktable.dir/common/cache.c.o
/home/garrett/Source/darktable/src/common/bilateral.c: In function ‘dt_bilateral_splat’:
/home/garrett/Source/darktable/src/common/bilateral.c:138:25: error: ‘in’ not specified in enclosing ‘parallel’
  138 |       const float L = in[index];
      |                         ^
/home/garrett/Source/darktable/src/common/bilateral.c:130:9: error: enclosing ‘parallel’
  130 | #pragma omp parallel for default(none) shared(b)
      |         ^~~
/home/garrett/Source/darktable/src/common/bilateral.c:154:75: error: ‘oy’ not specified in enclosing ‘parallel’
  154 |         const size_t ii = grid_index + ((k & 1) ? ox : 0) + ((k & 2) ? oy : 0) + ((k & 4) ? oz : 0);
      |                                                             ~~~~~~~~~~~~~~^~~~
/home/garrett/Source/darktable/src/common/bilateral.c:130:9: error: enclosing ‘parallel’
[...]
```
(and there are many more OpenMP-related errors, but I don't want to flood this bug report)

Here's what happens with `./build.sh --prefix /usr/local --build-type Release --disable-openmp`:

```
[...]
In file included from /home/garrett/Source/darktable/src/common/color_picker.c:20:
In function ‘dt_unreachable_codepath_with_caller’,
    inlined from ‘dt_color_picker_helper’ at /home/garrett/Source/darktable/src/common/color_picker.c:407:5:
/home/garrett/Source/darktable/src/common/darktable.h:549:3: error: ‘%s’ directive argument is null [-Werror=format-overflow=]
  549 |   fprintf(stderr, "[dt_unreachable_codepath] {%s} %s:%d (%s) - we should not be here. please report this to "
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  550 |                   "the developers.",
      |                   ~~~~~~~~~~~~~~~~~~
  551 |           description, file, line, function);
      |           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
[...]
```

---

I'm happy to try anything you suggest, including compiling from any branches to test out.

Thanks in advance!

@berndf: Thanks for working on this!

---

Here's what I get with `./build.sh --prefix /usr/local --build-type Release` when trying to compile on my Fedora 30 machine today using your `gcc9` branch:

```
[ 25%] Building C object src/CMakeFiles/lib_darktable.dir/common/colorspaces.c.o
/home/garrett/Source/darktable/src/common/color_picker.c: In function ‘color_picker_helper_4ch_parallel’:
/home/garrett/Source/darktable/src/common/color_picker.c:104:19: error: ‘cst_to’ not specified in enclosing ‘parallel’
  104 |         if(cst_to == iop_cs_LCh)
      |            ~~~~~~~^~~~~~~~~~~~~
/home/garrett/Source/darktable/src/common/color_picker.c:86:9: error: enclosing ‘parallel’
   86 | #pragma omp parallel default(none) shared(mean,mmin,mmax,width,box,pixel,w)
      |         ^~~
gmake[2]: *** [src/CMakeFiles/lib_darktable.dir/build.make:181: src/CMakeFiles/lib_darktable.dir/common/color_picker.c.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
gmake[1]: *** [CMakeFiles/Makefile2:1482: src/CMakeFiles/lib_darktable.dir/all] Error 2
gmake: *** [Makefile:152: all] Error 2
```

--- a/src/develop/develop.c
+++ b/src/develop/develop.c
@@ -1065,7 +1065,7 @@ void dt_dev_reload_history_items(dt_develop_t *dev)
   dt_dev_reorder_gui_module_list(dev);
 
   // we update show params for multi-instances for each other instances
-  //dt_dev_modules_update_multishow(dev);
+  dt_dev_modules_update_multishow(dev);
 }
 
 void dt_dev_pop_history_items_ext(dt_develop_t *dev, int32_t cnt)

